packages/strapi-design-system/src/Grid/Grid.tsx

```diff
@@ -1,19 +1,23 @@
 import * as React from 'react';
 
-import PropTypes from 'prop-types';
 import styled from 'styled-components';
 
 import { GridContext } from './GridContext';
-import { Box } from '../Box';
+import { Box, BoxProps } from '../Box';
 import handleResponsiveValues from '../helpers/handleResponsiveValues';
 
-const GridWrapper = styled(Box)`
+export interface GridProps extends BoxProps {
+  gridCols?: number;
+  gap?: number | number[];
```

Comment:
Do you think we could create a ResponsiveValue type that will be used in all the responsive properties? We could have a stronger typing by using the `theme.spaces` values as well 🤷🏽‍♂️ .

The `Proptypes` of this component weren't updated I think because responsive values can also be set as an object `{ desktop: ..., mobile: ..., tablet: ... }`.

Reply:
something like `keyof DefaultTheme['spaces'] | Array<keyof DefaultTheme['spaces']>` similar to what we use in `Flex` could be an option?

Reply:
Yes, it could be. We'll need to handle the object type as well `{ desktop: ..., mobile: ..., tablet: ... }`

Reply:
Could even do a named `tuple` too 👍🏼